server/mailet/mailets/src/test/java/org/apache/james/transport/mailets/SubAddressingTest.java

```diff
@@ -96,40 +97,79 @@ void shouldNotAddStorageDirectiveWhenNoRight() throws Exception {
     @Test
     void shouldAddStorageDirectiveMatchingDetailsWhenRight() throws Exception {
 
-        MailboxSession session = mailboxManager.createSystemSession(Username.of("recipient"));
+        Username username = Username.of("recipient");
+        MailboxSession session = mailboxManager.createSystemSession(username);
 
         // create Mailbox
         MailboxId mailboxId = mailboxManager.createMailbox(
-            MailboxPath.forUser(Username.of("recipient"), "any"), session)
-            .get();
+            MailboxPath.forUser(username, "any"), session).get();
 
-        // give right to post
-        MailboxACL.ACLCommand command = MailboxACL.command()
-            .key(MailboxACL.ANYBODY_KEY)
-            .rights(MailboxACL.Right.Post)
-            .asAddition();
+        giveRightToPost(mailboxId, session);
 
-        mailboxManager.applyRightsCommand(mailboxId, command, session);
+        Mail mail = sendSubAddressedMail(true);
+
+        AttributeName recipient = AttributeName.of("DeliveryPaths_recipient@localhost");
+        assertThat(mail.attributes().map(this::unbox))
+            .doesNotContain(Pair.of(recipient, "any"));
+    }
+
+    @Test
+    void shouldAddStorageDirectiveWhenEveryoneHasRightAndSenderIsUnknown() throws Exception {
+
+        Username username = Username.of("recipient");
+        MailboxSession session = mailboxManager.createSystemSession(username);
+
+        // create Mailbox
+        MailboxId mailboxId = mailboxManager.createMailbox(
+                MailboxPath.forUser(username, "any"), session).get();
+
+        giveRightToPost(mailboxId, session);
+        Mail mail = sendSubAddressedMail(false);
+
+        AttributeName recipient = AttributeName.of("DeliveryPaths_recipient@localhost");
+        assertThat(mail.attributes().map(this::unbox))
+            .doesNotContain(Pair.of(recipient, "any"));
+    }
+
+    @Test
+    void shouldNotAddStorageDirectiveWhenNooneHasRightAndSenderIsUnknown() throws Exception {
+
+        Username username = Username.of("recipient");
+        MailboxSession session = mailboxManager.createSystemSession(username);
+
+        // create Mailbox
+        MailboxId mailboxId = mailboxManager.createMailbox(
+            MailboxPath.forUser(username, "any"), session).get();
 
-        Mail mail = sendSubAddressedMail();
+        giveRightToPost(mailboxId, session);
+        Mail mail = sendSubAddressedMail(false);
 
         AttributeName recipient = AttributeName.of("DeliveryPaths_recipient@localhost");
         assertThat(mail.attributes().map(this::unbox))
             .doesNotContain(Pair.of(recipient, "any"));
     }
 
-    private Mail sendSubAddressedMail() throws MessagingException, IOException {
+    private void giveRightToPost(MailboxId mailboxId, MailboxSession session) throws MailboxException {
```

Comment:
If we make MailboxACLCommand a constant then giveRightToPost method can likely be removed as it would add nothing.

Reply:
fair point ! 
the only thing is that initializing the ACLCommand throws an exception so i cannot initialize it as a static final constant directly. so i used a @BeforeAll method to initialize the different commands i will have to use, is that ok ?

Reply:
Yes IMO ok